snuba/clusters/cluster.py

```diff
@@ -1,15 +1,55 @@
 from abc import ABC, abstractmethod
-from typing import Any, Callable, Generic, Mapping, Optional, Set, TypeVar
+from enum import Enum
+from typing import (
+    Any,
+    Callable,
+    Generic,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Set,
+    Tuple,
+    TypeVar,
+)
 
 from snuba import settings
 from snuba.clickhouse.http import HTTPBatchWriter
 from snuba.clickhouse.native import ClickhousePool, NativeDriverReader
 from snuba.clickhouse.sql import SqlQuery
 from snuba.clusters.storage_sets import StorageSetKey
+from snuba.optimize import OPTIMIZE_SEND_RECEIVE_TIMEOUT
```

Comment:
I don't think the cluster (which is a low level abstraction should depend on the optimize process.  At best the opposite should be true. This seems a settings of one specific use case of clickhouse client settings (which is here). What about defining it here and make optimize depend on this instead?

Reply:
Sounds good, moved it into the client settings now.